homeassistant/components/climate/modbus.py

```diff
@@ -1,148 +1,410 @@
+        mod = config.get(prop)
+        if not mod:
+            continue
+
+        count = mod[CONF_COUNT] if CONF_COUNT in mod else 1
+        data_type = mod.get(CONF_DATA_TYPE)
+        if data_type != DATA_TYPE_CUSTOM:
+            try:
+                mod[CONF_STRUCTURE] = '>{}'.format(data_types[
+                    DATA_TYPE_INT if data_type is None else data_type][count])
+            except KeyError:
+                _LOGGER.error("Unable to detect data type for %s", prop)
+                continue
+
+        try:
+            size = struct.calcsize(mod[CONF_STRUCTURE])
+        except struct.error as err:
+            _LOGGER.error(
+                "Error in sensor %s structure: %s", prop, err)
+            continue
+
+        if count * 2 != size:
+            _LOGGER.error(
+                "Structure size (%d bytes) mismatch registers count "
+                "(%d words)", size, count)
+            continue
+
+        mods[prop] = mod
+
+    if not mods:
+        _LOGGER.error("Invalid config %s: no modbus items", name)
+        return
+
+    def has_valid_register(mods, index):
+        """Check valid register."""
+        for prop in mods:
+            registers = mods[prop].get(CONF_REGISTERS)
+            if not registers or index >= len(registers):
+                return False
+        return True
+
+    devices = []
+    for index in range(100):
+        if not has_valid_register(mods, index):
+            break
+        devices.append(ModbusClimate(name, operation_list, fan_list,
+                                     swing_list, mods, index))
+
+    if not devices:
+        for prop in mods:
+            if CONF_REGISTER not in mod:
+                _LOGGER.error("Invalid config %s/%s: no register", name, prop)
+                return
+        devices.append(ModbusClimate(name, operation_list, fan_list,
+                                     swing_list, mods))
+
+    add_devices(devices, True)
+
+
+class ModbusClimate(ClimateDevice):
+    """Representation of a Modbus climate device."""
+
+    def __init__(self, name, operation_list,
+                 fan_list, swing_list, mods, index=-1):
+        """Initialize the climate device."""
+        self._name = name + str(index + 1) if index != -1 else name
+        self._index = index
+        self._mods = mods
+        self._operation_list = operation_list
+        self._fan_list = fan_list
+        self._swing_list = swing_list
+        self._values = {}
 
     @property
     def name(self):
         """Return the name of the climate device."""
         return self._name
 
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        features = 0
+        for prop in self._mods:
+            features |= SUPPORTED_FEATURES[prop]
+        return features
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self.unit_of_measurement
```

Comment:
Is it possible to reference this here? self.unit_of_measurement is not defined.

Reply:
This needs to fetch the unit of what is used to fetch current temp etc